src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/Main.cs

```diff
@@ -0,0 +1,214 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using ManagedCommon;
+using Microsoft.PowerToys.Run.Plugin.WindowsRegistry.Helper;
+using Microsoft.PowerToys.Run.Plugin.WindowsSettings.Helper;
+using Microsoft.PowerToys.Run.Plugin.WindowsSettings.Properties;
+using Wox.Plugin;
+
+namespace Microsoft.PowerToys.Run.Plugin.WindowsSettings
+{
+    /// <summary>
+    /// Main class of this plugin that implement all used interfaces.
+    /// </summary>
+    public class Main : IPlugin, IContextMenu, IPluginI18n, IDisposable
+    {
+        /// <summary>
+        /// The path to the symbol for a light theme.
+        /// </summary>
+        private const string _lightSymbol = "Images/WindowsSettings.light.png";
+
+        /// <summary>
+        /// The path to the symbol for a dark theme.
+        /// </summary>
+        private const string _darkSymbol = "Images/WindowsSettings.dark.png";
+
+        /// <summary>
+        /// The name of this assembly.
+        /// </summary>
+        private readonly string _assemblyName;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data).
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result.
+        /// </summary>
+        private string _defaultIconPath;
+
+        /// <summary>
+        /// Indicate that the plugin is disposed.
+        /// </summary>
+        private bool _disposed;
+
+        /// <summary>
+        /// List that contain all settings.
+        /// </summary>
+        private IEnumerable<WindowsSetting>? _settingsList;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            _assemblyName = Assembly.GetExecutingAssembly().GetName().Name ?? Name;
+            _defaultIconPath = _lightSymbol;
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>.
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin.</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+
+            var settingsList = ResourceHelper.ReadAllPossibleSettings();
+
+            TranslationHelper.TranslateAllSettings(settingsList);
+
+            _settingsList = WindowsVersionHelper.FilterByVersion(settingsList);
```

Comment:
```suggestion
            WindowsVersionHelper.FilterByVersion(settingsList);

            TranslationHelper.TranslateAllSettings(settingsList);
```

Let's change order (and remove var `_SettingsList`) to save time when translating. With this change we only translate the available settings and not all settings.